src/urllib3/util/retry.py

```diff
@@ -306,6 +307,8 @@ def _is_connection_error(self, err):
         """ Errors when we're fairly sure that the server did not receive the
         request, so it should be safe to retry.
         """
+        if isinstance(err, ProxyError):
```

Comment:
As a part of this work can we amend the signature of `ProxyError` to be `__init__(self, message, error, *args)`? That way we're guaranteed to unpack this properly.

Reply:
Done with `err.orignal_error`.